### PR TITLE
toolbar: rework overflow css logic

### PR DESCRIPTION
**Pull request by mimecuvalo**
Small side quest: This reworks the overflow toolbar css to rely on `nth-child` instead of putting together a long selector.
This also address the minor issue/edge case raised in https://github.com/tldraw/tldraw/pull/3757

### Change Type

- [x] `sdk` — Changes the tldraw SDK
- [ ] `dotcom` — Changes the tldraw.com web app
- [ ] `docs` — Changes to the documentation, examples, or templates.
- [ ] `vs code` — Changes to the vscode plugin
- [ ] `internal` — Does not affect user-facing stuff

- [ ] `bugfix` — Bug fix
- [ ] `feature` — New feature
- [x] `improvement` — Improving existing features
- [ ] `chore` — Updating dependencies, other boring stuff
- [ ] `galaxy brain` — Architectural changes
- [ ] `tests` — Changes to any test code
- [ ] `tools` — Changes to infrastructure, CI, internal scripts, debugging tools, etc.
- [ ] `dunno` — I don't know

### Release Notes

- Toolbar: cleanup overflow css rules.

### CODE REVIEW

**Comment by steveruizok** [comment]
Moving reviewer to @somehats, who wrote the last implementation here

**Comment by mimecuvalo** [comment]
> Moving reviewer to @SomeHats, who wrote the last implementation here

ah thanks! for some reason i thought it was Lu but now i'm looking at the history and it was Alex 🙃

**Comment by mimecuvalo** [comment]
@steveruizok i copied over your re-order to this PR. do you think the arrows could all be in the same row though for the overflow menu?

seems odd that the last arrow is on a row by itself:
Screenshot 2024-05-21 at 09 09 26

File: packages/tldraw/src/lib/ui/components/Toolbar/OverflowingToolbar.tsx [hunk 1]
```diff
@@ -31,25 +31,13 @@ export function OverflowingToolbar({ children }: { children: React.ReactNode })
...
-
-		for (let i = 0; i < overflowIndex; i++) {
-			showInMainSelectors.push(`:nth-child(${i + 1})`)
-			hideFromOverflowSelectors.push(`:nth-child(${i + 1})`)
-		}
+		const activeCss = lastActiveOverflowItem ? `:not([data-value="${lastActiveOverflowItem}"])` : ''
 
 		return `
-			#${id}_main > *:not(${showInMainSelectors.join(', ')}) {
+			#${id}_main > *:nth-child(n + ${overflowIndex + (lastActiveOverflowItem ? 1 : 2)})${activeCss} {
```
Comment (SomeHats) [inline packages/tldraw/src/lib/ui/components/Toolbar/OverflowingToolbar.tsx:37]: TIL! this is so much better, ty

**Review by SomeHats — APPROVED** [review]
looks good, although i think we should try and keep the shapes in a nice block in the overflow

File: packages/tldraw/src/lib/ui/components/Toolbar/DefaultToolbarContent.tsx [hunk 2]
```diff
@@ -16,14 +16,15 @@ export function DefaultToolbarContent() {
...
 			<EllipseToolbarItem />
-			<DiamondToolbarItem />
 			<TriangleToolbarItem />
-			<TrapezoidToolbarItem />
-			<RhombusToolbarItem />
-			<HexagonToolbarItem />
+			<DiamondToolbarItem />
 			<CloudToolbarItem />
 			<StarToolbarItem />
+			<PentagonToolbarItem />
```
[the remark this answers is not in the document]
Comment (mimecuvalo) [reply packages/tldraw/src/lib/ui/components/Toolbar/DefaultToolbarContent.tsx:23]: i'd agree - @steveruizok you wanna rework this bit? not sure what you wanna cull

**Pull request merged**